modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/impl/TaskServiceImpl.java

```diff
@@ -192,7 +195,16 @@ public ActionEvent getActionEventFor(TaskActionInformation taskActionInformation
                 public List<Task> execute(Query query, InstanceSecurityRestriction restriction) {
                     String byTriggerSubject = "trigger.subject == param";
                     String isTaskActive = "enabled == true";
-                    String filter = String.format("(%s) && (%s)", isTaskActive, byTriggerSubject);
+                    String isNotUsingTimeWindow = "useTimeWindow != true";
+                    String now = (new Time(LocalTime.now(DateTimeZone.UTC))).toString();
+                    String isAfterStartTime = String.format("startTime < \"%s\"", now);
+                    String isBeforeEndTime = String.format("endTime > \"%s\"", now);
+                    String isEndTimeNextDay = "endTime < startTime";
+                    String isBetweenStartTimeAndMidnight = String.format("(%s) && (\"24:00\" > \"%s\")", isAfterStartTime, now);
+                    String isBetweenMidnightAndEndTime = String.format("(%s) && (\"00:00\" < \"%s\")", isBeforeEndTime, now);
+                    String isInTimeWindow = String.format("(%s) || ((%s) && (%s)) || ((%s) && ((%s) || (%s)))",
+                            isNotUsingTimeWindow, isAfterStartTime, isBeforeEndTime, isEndTimeNextDay, isBetweenStartTimeAndMidnight, isBetweenMidnightAndEndTime);
+                    String filter = String.format("(%s) && (%s) && (%s)", isTaskActive, byTriggerSubject, isInTimeWindow);
 
```

Comment:
get list of task from sql and then parse this list, it should be simplier than doing this using raw sql